src/spec.rs

```diff
@@ -342,92 +343,16 @@ impl Zip32CentralDirectoryEnd {
-
-            /* Identify how many bytes to read (this may be less than the window size for files
-             * smaller than END_WINDOW_SIZE). */
-            let end = (window_start + END_WINDOW_SIZE as u64).min(file_length);
-            let cur_len = (end - window_start) as usize;
-            debug_assert!(cur_len > 0);
-            debug_assert!(cur_len <= END_WINDOW_SIZE);
-            let cur_window: &mut [u8] = &mut window[..cur_len];
-            /* Read the window into the bytes! */
-            reader.read_exact(cur_window)?;
-
-            /* Find instances of the magic signature. */
-            for offset in finder.rfind_iter(cur_window) {
-                let cde_start_pos = window_start + offset as u64;
-                reader.seek(io::SeekFrom::Start(cde_start_pos))?;
-                /* Drop any headers that don't parse. */
-                if let Ok(cde) = Self::parse(reader) {
-                    results.push((Rc::new(cde), cde_start_pos));
-                }
-            }
-
-            /* We always want to make sure we go allllll the way back to the start of the file if
-             * we can't find it elsewhere. However, our `while` condition doesn't check that. So we
-             * avoid infinite looping by checking at the end of the loop. */
-            if window_start == search_lower_bound {
-                break;
-            }
-            /* Shift the window by END_WINDOW_SIZE bytes, but make sure to cover matches that
-             * overlap our nice neat window boundaries! */
-            window_start = (window_start
-                /* NB: To catch matches across window boundaries, we need to make our blocks overlap
-                 * by the width of the pattern to match. */
-                + mem::size_of::<Magic>() as u64)
-                /* This should never happen, but make sure we don't go past the end of the file. */
-                .min(file_length);
-            window_start = window_start
-                .saturating_sub(
-                    /* Shift the window upon each iteration so we search END_WINDOW_SIZE bytes at
-                     * once (unless limited by file_length). */
-                    END_WINDOW_SIZE as u64,
-                )
-                /* This will never go below the value of `search_lower_bound`, so we have a special
-                 * `if window_start == search_lower_bound` check above. */
-                .max(search_lower_bound);
-        }
-        if results.is_empty() {
-            Err(ZipError::InvalidArchive(
-                "Could not find central directory end",
-            ))
-        } else {
-            Ok(results.into_boxed_slice())
-        }
-    }
-
     pub fn write<T: Write>(self, writer: &mut T) -> ZipResult<()> {
         let (block, comment) = self.block_and_comment()?;
         block.write(writer)?;
         writer.write_all(&comment)?;
         Ok(())
     }
+
+    pub fn is_zip64(&self) -> bool {
```

Comment:
Call this `may_be_zip64` instead, because a ZIP32 file may happen to have u16::MAX files or u32::MAX bytes before the central directory.